File: server-scala-play-server/app/api/TwitterApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.ProfileResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait TwitterApi {
  /**
    * Authorize Twitter
    * Makes an authorization call to twitter for a user to login and allow any app permissions.
    * @param appKey the application key
    */
  def authorizeTwitter(version: BigDecimal, appKey: String): SirqulResponse

  /**
    * Login Twitter
    * Returns the user profile information given an access token and the secret access token. This call verifies the tokens with twitter and creates a Sirqul account for the user if its their first time logging in.
    * @param accessToken The access token
    * @param accessTokenSecret The secret access token
    * @param appKey The application key
    * @param responseFilters a comma separated list of ProfileFilters for filtering the returned response data
    * @param deviceId The device id
    * @param latitude The current latitude of the user
    * @param longitude The current longitude of the user
    */
  def loginTwitter(version: BigDecimal, accessToken: String, accessTokenSecret: String, appKey: String, responseFilters: String, deviceId: Option[String], latitude: Option[Double], longitude: Option[Double]): ProfileResponse
}

File: server-ruby-sinatra/api/twitter_api.rb
require 'json'


MyApp.add_route('POST', '/api/3.18/twitter/authorize', {
  "resourcePath" => "/Twitter",
  "summary" => "Authorize Twitter",
  "nickname" => "authorize_twitter",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/twitter/authorize",
  "notes" => "Makes an authorization call to twitter for a user to login and allow any app permissions.",
  "parameters" => [
    {
      "name" => "app_key",
      "description" => "the application key",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/twitter/login', {
  "resourcePath" => "/Twitter",
  "summary" => "Login Twitter",
  "nickname" => "login_twitter",
  "responseClass" => "ProfileResponse",
  "endpoint" => "/twitter/login",
  "notes" => "Returns the user profile information given an access token and the secret access token. This call verifies the tokens with twitter and creates a Sirqul account for the user if its their first time logging in.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "access_token",
      "description" => "The access token",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "access_token_secret",
      "description" => "The secret access token",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "The application key",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "response_filters",
      "description" => "a comma separated list of ProfileFilters for filtering the returned response data",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "The current latitude of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "The current longitude of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

